### PR TITLE
Comprehensive tests for polkadot-network

### LINKED ISSUE #121: Comprehensive tests for polkadot-network
1. That `TableRouter` actually routes messages and behaves as expected
2. That `ConsensusNetwork` actually works as expected and can manage multiple consensus instances.

### PR DESCRIPTION
**Pull request by gterzian**
FIX #121

### CODE REVIEW

**Comment by gterzian** [comment]
Putting this one on-ice, until the networking code settles down into it's new configuration...

File: network/src/tests/mod.rs [hunk 1]
```diff
@@ -20,20 +20,98 @@ use super::{PolkadotProtocol, Status, Message, FullStatus};
...
+		PolkadotProtocol::new(None)
+	}
+}
+
+pub struct ForceFinalized(Arc<PeersClient>);
+
+impl JustificationImport<Block> for ForceFinalized {
+	type Error = ConsensusError;
+
+	fn import_justification(
```
Comment (ltfschoen) [inline network/src/tests/mod.rs]: these arguments could fit on a single line (it's no longer than https://github.com/paritytech/polkadot/pull/202/files#diff-383209c932e7187683ba4eea46d5b30dR95)

**Comment by bkchr** [comment]
@vs77bb this issue should still be available.

**Comment by kianenigma** [comment]
is this stale or will it be worked on by anyone?

**Comment by thiolliere** [comment]
also @amaurymartiny told me about chaos network test in javascript, is that related ?

**Comment by bkchr** [comment]
Probably on a more high level, but we need more integration tests.

**Comment by thiolliere** [comment]
for information https://gitlab.parity.io/parity/infrastructure/parity-testnet/

**Comment by amaurymartiny** [comment]
We had a small chat some time ago about whether the chaos-net should do e2e tests like these ones, and I don't think it would fit here.

Reasons:
- the chaos net handles stuff like managing Kubernetes nodes (via the Kubernetes sdk), and I don't think putting such code in this repo would be a good idea (rust core devs might not all be interested in writing devops-related code).
- the chaos net is all written in JS (because right now polkadotjs api is the best way to communicate with substrate metadata-based chains). I don't think we should put JS code in this repo.
- overall the chaos net will be an independent project with a project owner, so for now everything chaos-net lives in its own repo (the one Guillaume put just above).

Once the chaos net stabilizes a bit, there might be a possibility to add to offer a way for these PRs to spawn an E2E test in the chaos net (or in a chaos-net clone). But for the near future that is not happening.

**Pull request closed without merging**